src/rpc/methods/eth.rs

```diff
@@ -1705,6 +1773,73 @@ impl RpcMethod<0> for EthProtocolVersion {
     }
 }
 
+pub enum EthGetTransactionByHash {}
+impl RpcMethod<1> for EthGetTransactionByHash {
+    const NAME: &'static str = "Filecoin.EthGetTransactionByHash";
+    const NAME_ALIAS: Option<&'static str> = Some("eth_getTransactionByHash");
+    const PARAM_NAMES: [&'static str; 1] = ["tx_hash"];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (Hash,);
+    type Ok = Option<ApiEthTx>;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (tx_hash,): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let message_cid = if let Some(cid) = ctx.chain_store().get_mapping(&tx_hash)? {
```

Comment:
nit: this could potentially be more eloquent, but it's a matter of taste.

e.g.

```rust

let mut message_cid = tx_hash.to_cid();
if let Some(cid) ... {
  message_cid = ..
}
```

Reply:
Hmm. I do think it would be better without a `mut`.